rllib/algorithms/iql/iql_learner.py

```diff
@@ -0,0 +1,51 @@
+from typing import Dict
+
+from ray.rllib.algorithms.dqn.dqn_learner import DQNLearner
+from ray.rllib.utils.annotations import (
+    override,
+    OverrideToImplementCustomLogic_CallToSuperRecommended,
+)
+from ray.rllib.utils.lambda_defaultdict import LambdaDefaultDict
+from ray.rllib.utils.typing import ModuleID, TensorType
+
+QF_TARGET_PREDS = "qf_target_preds"
+VF_PREDS_NEXT = "vf_preds_next"
+VF_LOSS = "value_loss"
+
+
+class IQLLearner(DQNLearner):
+    @OverrideToImplementCustomLogic_CallToSuperRecommended
+    @override(DQNLearner)
+    def build(self) -> None:
+        # Define the expectile parameter(s).
+        self.expectile: Dict[ModuleID, TensorType] = LambdaDefaultDict(
+            lambda module_id: self._get_tensor_variable(
+                # Note, we want to train with a certain expectile.
+                [self.config.get_config_for_module(module_id).expectile],
+                trainable=False,
+            )
+        )
+
+        # Define the temperature for the acotr advantage loss.
+        self.temperature: Dict[ModuleID, TensorType] = LambdaDefaultDict(
+            lambda module_id: self._get_tensor_variable(
+                # Note, we want to train with a certain expectile.
+                [self.config.get_config_for_module(module_id).beta],
+                trainable=False,
+            )
+        )
+
+        # Store loss tensors here temporarily inside the loss function for (exact)
```

Comment:
super nit: Should we move this into `IQLTorchLearner` instead? This logic may be torch-specific.

Reply:
They are in `SACLearner` as well. I have no problem to move them over, but then we have to implement the `setup` method also for the `IQLTorchLearner` (I think the `expectile` and `temperature` could be defined in there, too - we might not even need `get_default_tensor` anymore b/c we are on Torch anyways - only if we are thinking about JAX or so in the future it makes sense to keep it).